clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusV2ClusterCachingAgent.java

```diff
@@ -77,10 +77,10 @@
   static final java.util.Set<AgentDataType> types =
       Collections.unmodifiableSet(
           Stream.of(
+                  AUTHORITATIVE.forType(CLUSTERS.ns),
```

Comment:
I don't think this is correct

The issue is that no single cluster caching agent can mark itself Authoritative for clusters because clusters span regions/accounts

We should really just have a caching agent that looks at all Server groups in cache and returns the authoritative list of clusters

What this change will cause is strange behaviour if you delete the last server group in a region (the whole cluster will disappear and reappear in cache).

Reply:
Yeah, that's what happens for Applications right now too :(

Reply:
FYI, if you look at the `SqlProviderCache` code that I deleted, this is already functionally true for SQL... this only changes the behavior for Redis users. And since you're the only Titus users and you only use SQL, I don't think this will actually affect anything in reality.

I actually spent a bunch of time several months ago trying to fix this behavior by refactoring CATS, but the lack of tests (and the fact that a ton of it is Groovy) meant I never could convince myself that it wasn't going to break other cloud providers, so I gave up :(